pkg/cmd/release-payload-controller/payload_rejected_controller.go

```diff
@@ -133,9 +157,64 @@ func (c *PayloadRejectedController) sync(ctx context.Context, key string) error
 		return err
 	}
 
+	originalCondition := conditions.FindCondition(originalReleasePayload.Status.Conditions, v1alpha1.ConditionPayloadRejected)
+	rejectedCondition := computeReleasePayloadRejectedCondition(originalReleasePayload, originalCondition)
+
 	releasePayload := originalReleasePayload.DeepCopy()
+	conditions.SetCondition(&releasePayload.Status.Conditions, rejectedCondition)
+	releasepayloadhelpers.CanonicalizeReleasePayloadStatus(releasePayload)
+
+	if reflect.DeepEqual(originalReleasePayload, releasePayload) {
+		return nil
+	}
 
 	klog.V(4).Infof("Syncing Payload Rejected for ReleasePayload: %s/%s", releasePayload.Namespace, releasePayload.Name)
+	_, err = c.releasePayloadClient.ReleasePayloads(releasePayload.Namespace).UpdateStatus(ctx, releasePayload, metav1.UpdateOptions{})
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
+
+func computeReleasePayloadRejectedCondition(payload *v1alpha1.ReleasePayload, originalCondition *metav1.Condition) metav1.Condition {
+	rejectedCondition := metav1.Condition{
+		Type:   v1alpha1.ConditionPayloadRejected,
+		Status: metav1.ConditionUnknown,
+		Reason: ReleasePayloadRejectedReason,
+	}
+
+	// Check for "Accepted" PayloadOverride
+	switch payload.Spec.PayloadOverride.Override {
+	case v1alpha1.ReleasePayloadOverrideRejected:
+		rejectedCondition.Status = metav1.ConditionTrue
+		rejectedCondition.Message = payload.Spec.PayloadOverride.Reason
+		rejectedCondition.Reason = ReleasePayloadManuallyRejectedReason
+		return rejectedCondition
+	case v1alpha1.ReleasePayloadOverrideAccepted:
+		rejectedCondition.Status = metav1.ConditionFalse
+		// Update previously Rejected condition
+		if originalCondition != nil {
```

Comment:
same comment here, don't use the old condition.

Reply:
Done!